lib/model/KnowledgePanelElement.dart

```diff
@@ -0,0 +1,204 @@
+
+  factory KnowledgePanelTextElement.fromJson(Map<String, dynamic> json) =>
+      _$KnowledgePanelTextElementFromJson(json);
+
+  @override
+  Map<String, dynamic> toJson() => _$KnowledgePanelTextElementToJson(this);
+}
+
+/// Image that represents the KnowledgePanel.
+@JsonSerializable()
+class KnowledgePanelImageElement extends JsonObject {
+  /// Url of the image.
+  final String url;
+
+  /// Width of the image.
+  ///
+  /// This is just a suggestion coming from the server, the client may choose to
+  /// use its own dimensions for the image.
+  final int? width;
+
+  /// Height of the image.
+  ///
+  /// This is just a suggestion coming from the server, the client may choose to
+  /// use its own dimensions for the image.
+  final int? height;
+
+  /// Alt Text of the image.
+  @JsonKey(name: 'alt_text')
+  final String? altText;
+
+  const KnowledgePanelImageElement(
+      {required this.url, this.width, this.height, this.altText});
+
+  factory KnowledgePanelImageElement.fromJson(Map<String, dynamic> json) =>
+      _$KnowledgePanelImageElementFromJson(json);
+
+  @override
+  Map<String, dynamic> toJson() => _$KnowledgePanelImageElementToJson(this);
+}
+
+/// Element representing a Panel Id of a KnowledgePanel. This element is a
+/// Knowledge panel itself, the KnowledgePanel can be found in the list of
+/// Knowledge panels using the id.
+@JsonSerializable()
+class KnowledgePanelPanelIdElement {
+  @JsonKey(name: 'panel_id')
+  final String panelId;
+
+  const KnowledgePanelPanelIdElement({required this.panelId});
+
+  factory KnowledgePanelPanelIdElement.fromJson(Map<String, dynamic> json) =>
+      _$KnowledgePanelPanelIdElementFromJson(json);
+
+  @override
+  Map<String, dynamic> toJson() => _$KnowledgePanelPanelIdElementToJson(this);
+}
+
+/// Element representing a tabular data for the KnowledgePanel.
+@JsonSerializable()
+class KnowledgePanelTableElement {
+  @JsonKey(name: 'table_id')
+  final String tableId;
+
+  @JsonKey(name: 'table_type')
+  final String tableType;
+
+  final String title;
+
+  final List<String> headers;
+
+  final List<List<String>> rows;
+
+  const KnowledgePanelTableElement(
+      {required this.tableId,
+      required this.tableType,
+      required this.title,
+      required this.headers,
+      required this.rows});
+
+  factory KnowledgePanelTableElement.fromJson(Map<String, dynamic> json) =>
+      _$KnowledgePanelTableElementFromJson(json);
+
+  @override
+  Map<String, dynamic> toJson() => _$KnowledgePanelTableElementToJson(this);
+}
+
+/// The type of Knowledge panel.
+enum KnowledgePanelElementType {
+  /// The description summarizes the knowledge panel.
+  @JsonValue('text')
+  TEXT,
+
+  @JsonValue('image')
+  IMAGE,
+
+  /// Disclaimer notes that the client may or may not choose to display.
+  @JsonValue('panel')
+  PANEL,
+
+  /// Disclaimer notes that the client may or may not choose to display.
+  @JsonValue('table')
```

Comment:
Server side I changed 'table' to 'table_2_columns', to clearly mark that it is expected to have only 2 columns. And we can have a more generic 'table' type with any number of columns.

Reply:
yeah, but can't we fit table_2_columns in the generic table ? client can easily figure out that there are two columns. if we want to pass extra info to the client, we can pass it in the type ("percent" in this case)?